Webhooks setup

Nithin-Varma opened the pull request:
**Database:** Made a webhooks table to store client settings and a webhook_deliveries table to track delivery attempts.

**Events:** Set up three main events: transaction.created (start), transaction.fiatConfirmed (fiat payment done), and transaction.onChainConfirmed (blockchain confirmed).

**Event System:** Built WebhookEventBusV1 to handle events separately from publishing them.

**Handler:** Created webhook-handler.ts to process events, find subscriptions, and send secure payloads to clients.

**Client API:** Added webhooks/index.ts with endpoints for clients to manage their webhooks.

**Publishing:** Linked bus.publish to checkout.ts (for starting transactions) and paypal-order-capture.ts (for fiat confirmation) and onchain confirmation is instant

**Efficiency:** Events only trigger if a project has active webhooks subscribed.

Nithin-Varma committed:
webhooks schema

Nithin-Varma committed:
Add webhook and webhook delivery tables with relationships for event handling

Nithin-Varma committed:
Merge branch 'webhooks' of https://github.com/normiedottech/normie.tech into webhooks

Nithin-Varma committed:
Remove webhooks table definition from database schema

dipanshuhappy committed:
Add webhook and webhook delivery tables with foreign key constraints

Nithin-Varma committed:
api endpoints for webhooks

Nithin-Varma committed:
webhook handler and integration

Nithin-Varma committed:
include transaction details in webhook

Nithin-Varma committed:
Add webhook event bus integration and enhance webhook handling logic

Nithin-Varma committed:
include transaction in properties

Nithin-Varma committed:
handled fiatconfirmation and onchain confirmation on paypal-order-capture, will see if we have better way to handle this.... i hope this is good

Nithin-Varma committed:
created to migrate

Nithin-Varma committed:
Improve webhook handler error logging and skip sending inactive webhooks

dipanshuhappy committed:
Refactor sst-env.d.ts files to remove redundant resource definitions and add reference paths for improved module resolution

dipanshuhappy commented:
@Nithin-Varma  So when i put example.com as the url , it should return an error and retry , that does not happen

dipanshuhappy commented:
@Nithin-Varma  basically test the max retries and the failures also if they are up to date

Nithin-Varma commented:
> @Nithin-Varma  basically test the max retries and the failures also if they are up to date
> 

I did, but I deleted the entries, when I changed the event from transaction.created to transaction.pending.

Nithin-Varma commented:
> @Nithin-Varma  So when i put example.com as the url , it should return an error and retry , that does not happen 

You mean while registering webhook?
If yes how could we know which is the correct URL.
It doesn't matter about the URL right? What all we need to do is send the post request to that URL and if they handle this post request their side they will receive payload.

If you mean if the url is not able to accept POST request it should retry and then return error.
I did test this case, let me check again.

Btw, make sure you are up to date with webhooks table..
There is no column called url, it's endpoint.

dipanshuhappy commented:
> > @Nithin-Varma  So when i put example.com as the url , it should return an error and retry , that does not happen
> 
> You mean while registering webhook? If yes how could we know which is the correct URL. It doesn't matter about the URL right? What all we need to do is send the post request to that URL and if they handle this post request their side they will receive payload.
> 
> If you mean if the url is not able to accept POST request it should retry and then return error. I did test this case, let me check again.
> 
> Btw, make sure you are up to date with webhooks table.. There is no column called url, it's endpoint.

If the post request fails ,it should retry , it wasn't retrying , yeah i am update to date webhook table

dipanshuhappy commented:
> > > @Nithin-Varma  So when i put example.com as the url , it should return an error and retry , that does not happen
> > 
> > 
> > You mean while registering webhook? If yes how could we know which is the correct URL. It doesn't matter about the URL right? What all we need to do is send the post request to that URL and if they handle this post request their side they will receive payload.
> > If you mean if the url is not able to accept POST request it should retry and then return error. I did test this case, let me check again.
> > Btw, make sure you are up to date with webhooks table.. There is no column called url, it's endpoint.
> 
> If the post request fails ,it should retry , it wasn't retrying , yeah i am update to date webhook table

I think its working i just tried it again
